sklearn/ensemble/_hist_gradient_boosting/histogram.pyx

```diff
@@ -103,7 +103,9 @@ cdef class HistogramBuilder:
 
     def compute_histograms_brute(
             HistogramBuilder self,
-            const unsigned int [::1] sample_indices):  # IN
+            const unsigned int [::1] sample_indices,
+            hist_struct [:, ::1] parent):  # IN
```

Comment:
I don't fully understand why you need to change the API here.

We have 2 routines to compute histograms:

- `compute_histograms_subtraction`, used when we know the histogram of the parent and the sibling of a node (using the subtraction trick)

- `compute_histograms_brute` used in all other cases. With that function, we do not need the histogram of the parent.